test/cpp/end2end/crl_provider_test.cc

```diff
@@ -220,6 +221,48 @@ TEST_F(CrlProviderTest, CrlProviderRevokedServer) {
   DoRpc(server_addr_, options, false);
 }
 
+TEST_F(CrlProviderTest, CrlProviderValidReloader) {
```

Comment:
(Can be deferred to a follow-up PR if you like) Should we have an e2e test where we try to reload from an empty directory?

Reply:
I think the edge cases are covered in unit tests for the directory reloader, I don't see much value in having e2e tests that test the same edge cases, what do you think?

Reply:
The reason I mention this here is that may be good to have an e2e test that ensures that the DynamicReloaderCrlProvider's GetCrl() method plays well with the code in ssl_transport_security.cc when it returns an either a missing or empty CRL. If you think the unit test coverage between DynamicReloaderCrlProvider and ssl_transport_security.cc is sufficient, that's totally fine to leave it, but just wanted to call this out as a place where we should ensure there is adequate testing.

Reply:
I suppose it can't hurt to expand the coverage - I'll make a not to myself, I don't think it belongs in this PR